vspace/src/main/java/edu/asu/diging/vspace/core/factory/impl/ImageBlockFactory.java

```diff
@@ -0,0 +1,32 @@
+package edu.asu.diging.vspace.core.factory.impl;
+
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.factory.IImageBlockFactory;
+import edu.asu.diging.vspace.core.model.IContentBlock;
+import edu.asu.diging.vspace.core.model.IImageBlock;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.IVSImage;
+import edu.asu.diging.vspace.core.model.impl.ImageBlock;
+
+@Service
+public class ImageBlockFactory implements IImageBlockFactory {
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * edu.asu.diging.vspace.core.factory.impl.IImageBlockFactory#createImageBlock(
+     * edu.asu.diging.vspace.core.model.ISlide,
+     * edu.asu.diging.vspace.core.model.IVSImage)
+     */
+    @Override
+    public IContentBlock createImageBlock(ISlide slide, IVSImage image) {
+        IContentBlock imageblock = new ImageBlock();
+        ((IImageBlock) imageblock).setImage(image);
+        imageblock.setSlide(slide);
+        imageblock.setDescription("image");
```

Comment:
why is that setting the description to "image"?

Reply:
Since ImageBlock properties are not accessible to TextBlock and vice-versa, when I was trying to display the blocks as ContentBlocks in a generalized way(using contentblock.text and contentblock.image.id), in the jsp page, I was getting the error. 
So, I wanted to identify them as text and image to distinguish and get only the required properties from those classes using "getAllContentBlocks(String slideId)".
Is there a way to access child properties using parent reference in jsp?